test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,675 @@
+              ",inclusive=" + std::string(inclusive ? "true" : "false");
+        }
+    };
+
+    const uint64_t MAX_ROLLBACKS = 100;
+    enum bounds { NO_BOUNDS, LOWER_BOUND_SET, UPPER_BOUND_SET, ALL_BOUNDS_SET };
+    bool reverse_collator_enabled = false;
+
+    public:
+    cursor_bound_01(const test_args &args) : test(args)
+    {
+        /* Track reverse_collator value as it is required for the custom comparator. */
+        reverse_collator_enabled = _config->get_bool(REVERSE_COLLATOR);
+    }
+
+    bool
+    custom_lexicographical_compare(
+      const std::string &first_key, const std::string &second_key, bool inclusive)
+    {
+        if (this->reverse_collator_enabled)
+            if (!inclusive)
+                return first_key.compare(second_key) > 0;
+            else
+                return first_key.compare(second_key) >= 0;
+        else if (!inclusive)
+            return first_key.compare(second_key) < 0;
+        else
+            return first_key.compare(second_key) <= 0;
+    }
+
+    void
+    cursor_traversal(scoped_cursor &range_cursor, scoped_cursor &normal_cursor,
+      const bound &lower_bound, const bound &upper_bound, bool next)
+    {
+        int exact, normal_ret, range_ret;
+        exact = normal_ret = range_ret = 0;
+        if (next) {
+            range_ret = range_cursor->next(range_cursor.get());
+            /*
+             * If the key exists, position the cursor to the lower key using search near otherwise
+             * use prev().
+             */
+            if (lower_bound.key.length() != 0) {
+                normal_cursor->set_key(normal_cursor.get(), lower_bound.key.c_str());
+                normal_ret = normal_cursor->search_near(normal_cursor.get(), &exact);
+                if (normal_ret == WT_NOTFOUND)
+                    return;
+                if (exact < 0)
+                    normal_ret = normal_cursor->next(normal_cursor.get());
+            } else
+                normal_ret = normal_cursor->next(normal_cursor.get());
+        } else {
+            range_ret = range_cursor->prev(range_cursor.get());
+            /*
+             * If the key exists, position the cursor to the upper key using search near otherwise
+             * use next().
+             */
+            if (upper_bound.key.length() != 0) {
+                normal_cursor->set_key(normal_cursor.get(), upper_bound.key.c_str());
+                normal_ret = normal_cursor->search_near(normal_cursor.get(), &exact);
+                if (normal_ret == WT_NOTFOUND)
+                    return;
+                if (exact > 0)
+                    normal_ret = normal_cursor->prev(normal_cursor.get());
+            } else
+                normal_ret = normal_cursor->prev(normal_cursor.get());
+        }
+
+        if (normal_ret == WT_NOTFOUND)
+            return;
+
+        testutil_assert(normal_ret == range_ret && (normal_ret == 0 || normal_ret == WT_NOTFOUND));
```

Comment:
Is it worth it checking for `normal_ret == WT_NOTFOUND` since you `return` just above if that's the case?

Reply:
Yeah, it is possible that the next() or prev() return back WT_NOTFOUND as well.

Reply:
I meant, can the assert become:
```
        testutil_assert(normal_ret == range_ret && normal_ret == 0);
```
Because we do this just before:
```
        if (normal_ret == WT_NOTFOUND)
            return;
```